pipelines/batch/src/main/java/com/google/fhir/analytics/JdbcResourceWriter.java

```diff
@@ -85,65 +97,92 @@ static void createTables(FhirEtlOptions options)
     log.info(
         String.format(
             "Connecting to DB url %s with user %s.", dbConfig.jdbcUrl(), dbConfig.dbUser()));
-    if (options.getUseSingleSinkTable()) {
-      // For CREATE statements we cannot (and don't need to) use a placeholder for table name, i.e.,
-      // we don't need to use PreparedStatement with '?'.
-      String tableCreate =
-          "CREATE TABLE IF NOT EXISTS %s (id VARCHAR(100) NOT NULL, "
-              + "type VARCHAR(50) NOT NULL, datab JSONB, PRIMARY KEY (id, type) );";
-      String createStatement = String.format(tableCreate, options.getSinkDbTablePrefix());
-      createSingleTable(jdbcSource, createStatement);
-    } else {
+
+    String viewDir = Strings.nullToEmpty(options.getViewDefinitionsDir());
+    if (viewDir.isEmpty()) {
+      log.info("Creating tables for each resource type.");
       for (String resourceType : options.getResourceList().split(",")) {
         String tableCreate =
-            "CREATE TABLE IF NOT EXISTS %s (id VARCHAR(100) NOT NULL, "
-                + "datab JSONB, PRIMARY KEY (id) );";
-        String createStatement =
-            String.format(tableCreate, options.getSinkDbTablePrefix() + resourceType);
+            "CREATE TABLE %s (id VARCHAR(100) NOT NULL, " + "datab JSONB, PRIMARY KEY (id) );";
+        String createStatement = String.format(tableCreate, resourceType);
         createSingleTable(jdbcSource, createStatement);
       }
+    } else {
+      ViewManager viewManager = ViewManager.createForDir(viewDir);
+      for (String resourceType : options.getResourceList().split(",")) {
+        ImmutableList<ViewDefinition> views = viewManager.getViewsForType(resourceType);
+        if (views == null || views.isEmpty()) {
+          log.warn("No views found for resource type {} in directory {}!", resourceType, viewDir);
+        } else {
+          for (ViewDefinition vDef : views) {
+            if (Strings.isNullOrEmpty(vDef.getName())) {
+              throw new ViewDefinitionException("Field `name` in ViewDefinition is not defined.");
+            }
+            StringBuilder builder = new StringBuilder("CREATE TABLE ");
```

Comment:
Same a the previous comment - `CREATE TABLE IF NOT EXISTS`

Reply:
Agreed, that the best implementation would be to check if the schema is consistent with the existing one and fail if it is not.
Also just a thought, should the views have `Primary Key`. Since there can be cross-joins, using the `Resource.id` might not be right. The [spec](https://build.fhir.org/ig/FHIR/sql-on-fhir-v2/StructureDefinition-ViewDefinition.html#getresourcekey--keytype) here says some ways of generating the Primary key. Bringing this up, so that we are aware of it.

Reply:
Yeah, that is the TODO on line 168 below, i.e., to add the requirement for having resource ID; we actually need that for other reasons too, e.g., updating old rows.